BinaryTree/Source/BinaryTree.cpp

```diff
@@ -0,0 +1,52 @@
+//
+// Created by Sam Petrov on 20.05.2020.
+//
+
+#include "BinaryTree.h"
+
+BinaryTree::BinaryTree(const int* arr, const int start, const int end) {
+    root = CreateMinimalBST(arr, start, end);
+}
+
+BinaryTree::~BinaryTree() {
+    delete root;
+}
+
+TreeNode* BinaryTree::search(int data) const {
+    return search(root, data);
+}
+
+TreeNode* BinaryTree::search(TreeNode* node, int data) {
+    if(node == nullptr || data == node->getData())
+        return node;
+    if(data < node->getData())
+        return search(node->getLeft(), data);
+    else
+        return search(node->getRight(), data);
+}
+
+void BinaryTree::insert(int d) {
+    insert(root, d);
+}
+
+TreeNode* BinaryTree::insert(TreeNode* node, int data) {
```

Comment:
Как-то замучено получилось... Можно же 
```suggestion
void BinaryTree::insert(TreeNode* &node, int data) {
```

Reply:
Какому из условий это не удовлетворяет?

Reply:
Такого не писал...
Это необязательно.

Reply:
Значит задача закрыта?

Reply:
Нет, ниже ещё замечания...